x11-packages/emacs-pgtk/build.sh

```diff
@@ -0,0 +1,143 @@
+# Prevent linking against libelf:
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS+=" ac_cv_lib_elf_elf_begin=no"
+# implemented using dup3(), which fails if oldfd == newfd
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS+=" gl_cv_func_dup2_works=no"
+# disable setrlimit function to make termux-am work from within emacs
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS+=" ac_cv_func_setrlimit=no"
+TERMUX_PKG_HOSTBUILD=true
+
+# Remove some irrelevant files:
+TERMUX_PKG_RM_AFTER_INSTALL="
+bin/grep-changelog
+share/applications/emacs.desktop
+share/emacs/${TERMUX_PKG_VERSION}/etc/emacs.desktop
+share/emacs/${TERMUX_PKG_VERSION}/etc/emacs.icon
+share/emacs/${TERMUX_PKG_VERSION}/etc/images
+share/emacs/${TERMUX_PKG_VERSION}/etc/refcards
+share/emacs/${TERMUX_PKG_VERSION}/etc/tutorials/TUTORIAL.*
+share/icons
+share/man/man1/grep-changelog.1.gz
+"
+
+# Remove ctags from the emacs package to prevent conflicting with
+# the Universal Ctags from the 'ctags' package (the bin/etags
+# program still remain in the emacs package):
+TERMUX_PKG_RM_AFTER_INSTALL+=" bin/ctags share/man/man1/ctags.1 share/man/man1/ctags.1.gz"
+
+termux_step_post_get_source() {
+	# Certain packages are not safe to build on device because their
+	# build.sh script deletes specific files in $TERMUX_PREFIX.
+	if $TERMUX_ON_DEVICE_BUILD; then
+		termux_error_exit "Package '$TERMUX_PKG_NAME' is not safe for on-device builds."
+	fi
+
+	# XXX: We have to start with new host build each time
+	#      to avoid build error when cross compiling.
+	rm -Rf $TERMUX_PKG_HOSTBUILD_DIR
+
+	# Termux only use info pages for emacs. Remove the info directory
+	# to get a clean Info directory file dir.
+	rm -Rf $TERMUX_PREFIX/share/info
+}
+
+termux_step_host_build() {
+	# Build a bootstrap-emacs binary to be used in termux_step_post_configure.
+	# Needed to compile a host arch compatible executable for el files
+	local NATIVE_PREFIX=$TERMUX_PKG_TMPDIR/emacs-native
+	mkdir -p $NATIVE_PREFIX/share/emacs/$TERMUX_PKG_VERSION
+	ln -s $TERMUX_PKG_SRCDIR/lisp $NATIVE_PREFIX/share/emacs/$TERMUX_PKG_VERSION/lisp
+	( cd $TERMUX_PKG_SRCDIR; ./autogen.sh )
+	$TERMUX_PKG_SRCDIR/configure --prefix=$NATIVE_PREFIX --without-all --without-x
+	make -j $TERMUX_MAKE_PROCESSES
+}
+
+termux_step_post_configure() {
+	cp $TERMUX_PKG_HOSTBUILD_DIR/src/bootstrap-emacs $TERMUX_PKG_BUILDDIR/src/bootstrap-emacs
+	cp $TERMUX_PKG_HOSTBUILD_DIR/lib-src/make-docfile $TERMUX_PKG_BUILDDIR/lib-src/make-docfile
+	cp $TERMUX_PKG_HOSTBUILD_DIR/lib-src/make-fingerprint $TERMUX_PKG_BUILDDIR/lib-src/make-fingerprint
+	cp -r $TERMUX_PKG_SRCDIR/lisp/* $TERMUX_PKG_BUILDDIR/lisp
+	cp -r $TERMUX_PKG_SRCDIR/etc $TERMUX_PKG_BUILDDIR
+	# Update timestamps so that the binaries does not get rebuilt:
+	touch -d "next hour" $TERMUX_PKG_BUILDDIR/src/bootstrap-emacs \
+		$TERMUX_PKG_BUILDDIR/lib-src/make-docfile \
+		$TERMUX_PKG_BUILDDIR/lib-src/make-fingerprint
+}
+
+termux_step_make() {
+	make -j $TERMUX_MAKE_PROCESSES
+    mkdir -p /data/data/com.termux/files/usr/share/emacs/master/lisp/emacs-lisp/
```

Comment:
I think you can move this line to `termux_step_post_make_install`, eliminating additional `termux_step_make` altogether. Or is this line a dependency for `make install`?

PS: I just now realised there is no `termux_step_post_make`

Also preferably use `$TERMUX_PREFIX` instead of `/data/data/com.termux/files/usr`

Reply:
This should be addressed with the latest push, please see: https://github.com/termux/termux-packages/compare/6d25bf570f4a119d1f1b9dc378ced108d1741519..f633789445a3f8d8b2ac41ab7ae8666e499c31eb